app/js/instances/createContainerController.js

```diff
@@ -0,0 +1,59 @@
+var _ = require('lodash');
+
+function createContainerController($scope, $routeParams, $location, dockerService, instanceModel) {
+    'use strict';
+    $scope.host = $routeParams.host;
+
+    dockerService.images.list($scope.host, null);
+
+    $scope.$listenTo(instanceModel, function () {
+      $scope.images = _.map(instanceModel.getImages(), function(image) {
+
+        image.Tags = _.filter(image.RepoTags, function(repoTag) {
+          return repoTag !== '<none>:<none>';
+        }).join(', ');
+
+        return image;
+      });
+
+      $scope.images = _.filter($scope.images, function(image) {
+        return image.Tags !== '';
+      });
+
+      $scope.selectedImage = $scope.images[0];
+    });
+
+    $scope.enviromentVariables = [];
+    $scope.enviromentInput = "";
+    $scope.cmdInput = "";
+
+    $scope.addEnviromentVar = function(event) {
+      if ((event.which !== 13 && event.type === "keypress") || $scope.enviromentInput === "") {
```

Comment:
Yeah I re-read your comment and was like, oh crap.  I understand where you're coming from, but I'm not sure ```ng-submit``` is the right thing for this.  I have it so this event triggers on ```ng-blur``` as well since the user could potentially forget to press enter on the input field.

I also tried ```ng-submit``` on the input tag and it was submitting the form despite the ```preventDefault```. :(